### PR TITLE
Live updates on details screen

### PR DESCRIPTION
**Pull request by guytepper**
todo: 

- [ ] Localizations
- [ ] Colorize station cards
- [ ] Await ride ID before allowing stop press

### CODE REVIEW

File: app/hooks/use-ride-progress/use-ride-route.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+/**
+ * This function is used to extract ride details from the route needed for API calls
+ */
+function getRouteDetails(route: RouteItem) {
+  const originId = route.trains[0].originStationId.toString()
+  const destinationId = route.trains[route.trains.length - 1].destinationStationId.toString()
+  const [date, time] = formatDateForAPI(route.departureTime)
+
+  return [originId, destinationId, date, time]
```
Comment (planecore) [inline app/hooks/use-ride-progress/use-ride-route.ts:83]: Way too many values for an array, use an object instead

File: app/hooks/use-ride-progress/use-ride-route.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  const queryClient = useQueryClient()
+
+  const [nextStationId, setNextStationId] = useState<number>(route.trains[0].originStationId)
+  const [delay, setDelay] = useState<number>(0)
+
+  const [originId, destinationId, date, time] = useMemo(() => getRouteDetails(route), [route])
+
+  const refetchRoute = async () => {
+    // fetch route - needed for getting delay information
+    const routes = await new RouteApi().getRoutes(originId, destinationId, date, time)
```
Comment (planecore) [inline app/hooks/use-ride-progress/use-ride-route.ts:26]: Don't create a new RouteApi object every time

File: app/screens/route-details/components/start-ride-button.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  tintColor: "white",
+  transform: [{ rotateY: isRTL ? "180deg" : "0deg" }],
+}
+
+interface StartRideButtonProps {
+  route: RouteItem
+  screenName: "routeDetails" | "activeRide"
+}
+
+export const StartRideButton = observer(function StartRideButton(props: StartRideButtonProps) {
```
Comment (planecore) [inline app/screens/route-details/components/start-ride-button.tsx:35]: Don't animate start ride button when the view appears. Do it only when the user starts/end the ride.

File: app/screens/route-list/route-list-screen.tsx [hunk 4]
```diff
@@ -30,7 +27,6 @@ export const RouteListScreen = observer(function RouteListScreen({ navigation, r
 
   // this is for the screen to re-render when the user navigates back from the route details screen
   // and the user has started a ride - so the route card will be highlighted
-  useRideRerender(ride, navigation)
```
Comment (planecore) [inline app/screens/route-list/route-list-screen.tsx:33]: `useRideRerender` is removed but its comment isn't

File: app/screens/route-details/components/live-ride-sheet.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        bottom: 0,
+      }}
+    >
+      <Text style={{ fontSize: 20, fontWeight: "bold", color: color.success }}>{progressText}</Text>
+      <StopButton
+        onPress={() => {
+          ride.stopRide(ride.id)
+
+          if (screenName === "activeRide") {
+            navigation.goBack()
```
Comment (planecore) [inline app/screens/route-details/components/live-ride-sheet.tsx:62]: doesn't work

File: app/hooks/use-ride-progress/get-stop-stations-status.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+  // build the stop stations status array
+  const stopStationsStatusArray: StopStationStatusItem[] = array.map((stopStation) => ({
+    top: "idle",
+    bottom: "idle",
+    stationId: stopStation.stationId,
+  }))
+
+  // convert the array to an object, so we could use the station id as a key
+  const stopStationsObject = convertStopStationsStatusArrayToObject(stopStationsStatusArray)
```
Comment (planecore) [inline app/hooks/use-ride-progress/get-stop-stations-status.ts:35]: you can use lodash `keyBy(stopStationsStatusArray, "stationId")` instead

**Review by planecore — COMMENTED** [review]
In addition, I'd like the active ride modal to popup when the user press the live activity/dynamic island

File: app/screens/route-details/components/live-ride-sheet.tsx (hunk 5 above)
Comment (guytepper) [reply app/screens/route-details/components/live-ride-sheet.tsx:62]: It does for me. Did you check the active ride from the home screen?

**Review by planecore — APPROVED** [review]

**Pull request merged**